bib/internal/progress/progress.go

```diff
@@ -0,0 +1,296 @@
+package progress
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/cheggaaa/pb/v3"
+	"github.com/mattn/go-isatty"
+	"github.com/sirupsen/logrus"
+
+	"github.com/osbuild/images/pkg/osbuild"
+)
+
+var (
+	osStderr         io.Writer = os.Stderr
+	isattyIsTerminal           = isatty.IsTerminal
+)
+
+// ProgressBar is an interfacs for progress reporting when there is
+// an arbitrary amount of sub-progress information (like osbuild)
+type ProgressBar interface {
+	// SetProgress sets the progress details at the given "level".
+	// Levels should start with "0" and increase as the nesting
+	// gets deeper.
+	SetProgress(level int, msg string, done int, total int) error
+	// The high-level message that is displayed in a spinner
+	// (e.g. "Building image foo")
+	SetPulseMsg(fmt string, args ...interface{})
```

Comment:
what about `SetPulseMsgf` to indicate that formatting is supported? It seems to be a pretty standard practice in Go.